typing/persistent_env.ml

```diff
@@ -467,18 +521,23 @@ let check_for_unset_parameters penv global =
            }))
     global.Global_module.hidden_args
 
-let rec global_of_global_name penv ~check name =
+let rec global_of_global_name penv ~check name ~allow_excess_args =
+  let load () =
+    let pn =
+      find_pers_name ~allow_hidden:true penv ~check name ~allow_excess_args
+    in
+    pn.pn_global
+  in
   match Hashtbl.find penv.globals name with
-  | { gn_global; _ } -> gn_global
-  | exception Not_found ->
-      let pn = find_pers_name ~allow_hidden:true penv check name in
-      pn.pn_global
+  | { gn_global = (global, Exact); _ } -> global
+  | { gn_global = (_, Approximate); _ } -> load ()
+  | exception Not_found -> load ()
```

Comment:
This seems to be the only place where Exact/Approximate affects behavior other than raising internally. But, does `global_of_global_name` have to return exact global_module? I looked at a few call sites and ain't sure.